File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cs
using System;

public class Solution {
    public int[] solution(int[] arr) {
        int length = arr.Length;
        int two = 1;
        while(true)
        {
            if(length <= two)
            {
                break;
            }
            two *= 2;
        }
        int[] answer = new int[two];
        for(int i = 0; i < answer.Length; i++)
        {
            if(i < arr.Length)
                answer[i] = arr[i];
            else
                answer[i] = 0;
        }
        return answer;
    }
}

File: 프로그래머스/0/181857. 배열의 길이를 2의 거듭제곱으로 만들기/배열의 길이를 2의 거듭제곱으로 만들기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr) {
    vector<int> answer(arr);
    int num = 1;
    while(arr.size() > num)
        num *= 2;
    
    for(int i = arr.size(); i < num; i++)
        answer.push_back(0);
    return answer;
}